### PR TITLE
Adding Global DB backup logic

### PR DESCRIPTION
**Pull request by pereyra-m**
|Related issue|
|---|
|#6149|

## Description

This PR moves the `PRAGMA journal_mode=DELETE;` SQL statement out of the the transaction and adds to Global DB the backup capability. 

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation
- [x] Package installation
- [x] Source upgrade
- [x] Package upgrade
- [ ] Review logs syntax and correct language

- [ ] Added unit tests (for new features)

### CODE REVIEW

**Comment by pereyra-m** [comment]
### Tests performed

#### Upgrade from 3.13.2 to 4.0

The `schema_global_upgrade_v1.sql` upgrades the database successfully, now that the `PRAGMA journal_mode=DELETE` is out of the transaction.

#### Global DB corruption

The **global.db** database is modified to simulate an error in the version read by `wdb_upgrade_global()`. This forces a backup creation of the old DB and a new and empty **global.db** is successfully created.

File: src/wazuh_db/schema_global.sql [hunk 1]
```diff
@@ -6,6 +6,8 @@
  * and/or modify it under the terms of GPLv2.
  */
 
+PRAGMA journal_mode=DELETE;
```
Comment (DProvinciani) [inline src/wazuh_db/schema_global.sql:9]: Is this necessary here? As per the discussion that I had with @vikman90 , this is necessary only on the upgrade.

Comment (pereyra-m) [reply src/wazuh_db/schema_global.sql:9]: This is true, the default mode is **DELETE** 
[https://sqlite.org/wal.html](https://sqlite.org/wal.html)

**Review by DProvinciani — APPROVED** [review]
Looks good!

**Pull request merged**